UUIDs being parsed from translators as translator ids

```
(+0000000): TranslatorsAndStylesController: load raw translators for all translators
[ERROR] Zotero(+0000353): TranslatorsAndStylesController: Gemeinsamer Bibliotheksverbund ISBN raw translator missing target
[ERROR] Zotero(+0000246): TranslatorsAndStylesController: can't read data from file:///private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/fe590e29-8052-4086-b9a9-6fcabdbae4ba - Error Domain=NSCocoaErrorDomain Code=260 "The file “fe590e29-8052-4086-b9a9-6fcabdbae4ba” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/fe590e29-8052-4086-b9a9-6fcabdbae4ba, NSUnderlyingError=0x2824dfc60 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
[ERROR] Zotero(+0000000): TranslatorsAndStylesController: can't read data from file:///private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/7be5ded9-87a9-4056-af72-d6c917125a29 - Error Domain=NSCocoaErrorDomain Code=260 "The file “7be5ded9-87a9-4056-af72-d6c917125a29” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/7be5ded9-87a9-4056-af72-d6c917125a29, NSUnderlyingError=0x2824dfd20 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
[ERROR] Zotero(+0000000): TranslatorsAndStylesController: can't read data from file:///private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/829e5da7-6ff0-4a60-82d3-b3246d00d9ca - Error Domain=NSCocoaErrorDomain Code=260 "The file “829e5da7-6ff0-4a60-82d3-b3246d00d9ca” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/829e5da7-6ff0-4a60-82d3-b3246d00d9ca, NSUnderlyingError=0x2824dfe10 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
[ERROR] Zotero(+0000000): TranslatorsAndStylesController: can't read data from file:///private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/a73e665e-a74e-46b7-8e6e-ae053d8b6f83 - Error Domain=NSCocoaErrorDomain Code=260 "The file “a73e665e-a74e-46b7-8e6e-ae053d8b6f83” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/private/var/mobile/Containers/Shared/AppGroup/4C1E867A-A0E5-48D5-B61B-0BF4E60BEB26/translators/a73e665e-a74e-46b7-8e6e-ae053d8b6f83, NSUnderlyingError=0x2824dfe70 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

For some reason the app seems to be trying to parse out any UUID that appears in a translator as a translatorID — e.g., `fe590e29-8052-4086-b9a9-6fcabdbae4ba` is from a URL in the Washington Post translator. I'm assuming this is some sort of regexp gone wrong, but obviously it should be properly [extracting the JSON](https://github.com/zotero/zotero/blob/07309d7c25efa4a9e6aa5353dfaf8ca8727c415c/chrome/content/zotero/xpcom/translation/translators.js#L319) at the top of translators, parsing it as JSON, and getting the `translatorID` property, not parsing the whole file for any UUID.

Not sure if `Gemeinsamer Bibliotheksverbund ISBN raw translator missing target` is related or an unrelated problem.

--- a/Zotero/Controllers/TranslatorsAndStylesController.swift
+++ b/Zotero/Controllers/TranslatorsAndStylesController.swift
@@ -75,7 +75,7 @@ final class TranslatorsAndStylesController {
     weak var coordinator: TranslatorsControllerCoordinatorDelegate?
     private lazy var uuidExpression: NSRegularExpression? = {
         do {
-            return try NSRegularExpression(pattern: #"[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}"#)
+            return try NSRegularExpression(pattern: #"setTranslator\(['"](?<uuid>[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12})['"]\)"#)
         } catch let error {
             DDLogError("TranslatorsAndStylesController: can't create uuid expression - \(error)")
             return nil
@@ -688,7 +688,7 @@ final class TranslatorsAndStylesController {
         guard let uuidRegex = self.uuidExpression else { return [] }
 
         let matches = uuidRegex.matches(in: code, options: [], range: NSRange(code.startIndex..., in: code))
-        return Set(matches.compactMap({ $0.substring(at: 0, in: code).flatMap(String.init) }))
+        return Set(matches.compactMap({ $0.substring(withName: "uuid", in: code).flatMap(String.init) }))
     }
 
     /// Finds `endIndex` of metadata part in translator file (translator file consists of json metadata and code).

--- a/Zotero/Extensions/RegularExpression+Extensions.swift
+++ b/Zotero/Extensions/RegularExpression+Extensions.swift
@@ -17,11 +17,27 @@ extension NSTextCheckingResult {
         return Range(self.range(at: index), in: string)
     }
 
+    /// Creates Swift `Range` from result for given group name in string.
+    /// - parameter name: Group name of matched range in this result.
+    /// - parameter string: String from which this result originates.
+    /// - returns: `Range` if index is in bounds, `nil` otherwise.
+    func swiftRange(withName name: String, in string: String) -> Range<String.Index>? {
+        return Range(self.range(withName: name), in: string)
+    }
+
     /// Creates substring from result at given index in string.
     /// - parameter index: Index of matched substring in this result.
     /// - parameter string: String from which this result originates.
     /// - returns: Substring if index is in bounds, `nil` otherwise.
     func substring(at index: Int, in string: String) -> Substring? {
         return self.swiftRange(at: index, in: string).flatMap({ string[$0] })
+    }
+
+    /// Creates substring from result for given group name in string.
+    /// - parameter name: Group name of matched substring in this result.
+    /// - parameter string: String from which this result originates.
+    /// - returns: Substring if index is in bounds, `nil` otherwise.
+    func substring(withName name: String, in string: String) -> Substring? {
+        return self.swiftRange(withName: name, in: string).flatMap({ string[$0] })
     }
 }

